pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -0,0 +1,310 @@
+		logrus.Infof("prometheus: %v", *prometheus.Spec.Replicas)
+		// check started or not
+		if *prometheus.Spec.Replicas == 0 {
+			logrus.Infof("start prometheus")
+			allStarted = false
+			prometheusCopy := prometheus.DeepCopy()
+			if replicas, err := strconv.Atoi(prometheus.Annotations[ReplicaStorageNetworkAnnotation]); err == nil {
+				*prometheusCopy.Spec.Replicas = int32(replicas)
+			}
+			delete(prometheusCopy.Annotations, ReplicaStorageNetworkAnnotation)
+
+			if _, err := h.prometheus.Update(prometheusCopy); err != nil {
+				logrus.Warnf("prometheus update error %v", err)
+			}
+		}
+	}
+
+	// check managedchart fleet-local/rancher-monitoring paused
+	if monitoring, err := h.managedChartCache.Get(FleetLocalNamespace, RancherMonitoring); err == nil {
+		logrus.Infof("Rancher Monitoring: %v", monitoring.Spec.Paused)
+		// check pause or not
+		if monitoring.Spec.Paused {
+			logrus.Infof("start rancher monitoring")
+			allStarted = false
+			monitoringCopy := monitoring.DeepCopy()
+			monitoringCopy.Spec.Paused = false
+			delete(monitoringCopy.Annotations, PausedStorageNetworkAnnotation)
+
+			if _, err := h.managedCharts.Update(monitoringCopy); err != nil {
+				logrus.Warnf("rancher monitoring error %v", err)
+			}
+		}
+	}
+
+	// check deployment cattle-monitoring-system/rancher-monitoring-grafana replica
+	if grafana, err := h.deploymentCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringGrafana); err == nil {
+		logrus.Infof("Grafana: %v", *grafana.Spec.Replicas)
+		// check started or not
+		if *grafana.Spec.Replicas == 0 {
+			logrus.Infof("start grafana")
+			allStarted = false
+			grafanaCopy := grafana.DeepCopy()
+			if replicas, err := strconv.Atoi(grafana.Annotations[ReplicaStorageNetworkAnnotation]); err == nil {
+				*grafanaCopy.Spec.Replicas = int32(replicas)
+			}
+			delete(grafanaCopy.Annotations, ReplicaStorageNetworkAnnotation)
+
+			if _, err := h.deployments.Update(grafanaCopy); err != nil {
+				logrus.Warnf("Grafana update error %v", err)
+			}
+		}
+	}
+
+	return allStarted
+}
+
+// check Pod status, if all pods are stopped, return true
+func (h *Handler) checkPodStatusAndStop() bool {
+	allStopped := true
+
+	// check prometheus cattle-monitoring-system/rancher-monitoring-prometheus replica
+	if prometheus, err := h.prometheusCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringPrometheus); err == nil {
+		logrus.Infof("prometheus: %v", *prometheus.Spec.Replicas)
+		// check stopped or not
+		if *prometheus.Spec.Replicas != 0 {
+			logrus.Infof("stop prometheus")
+			allStopped = false
+			prometheusCopy := prometheus.DeepCopy()
+			prometheusCopy.Annotations[ReplicaStorageNetworkAnnotation] = strconv.Itoa(int(*prometheus.Spec.Replicas))
+			*prometheusCopy.Spec.Replicas = 0
+
+			if _, err := h.prometheus.Update(prometheusCopy); err != nil {
+				logrus.Warnf("prometheus update error %v", err)
+			}
+		}
+	}
+
+	// check managedchart fleet-local/rancher-monitoring paused
```

Comment:
set "paused" at first when stopping;  unset "paused" at last when starting.

it is kind of lock between here and `fleet-agent`

Reply:
i didn't get it in this part.
Could you help me to describe more?

Reply:
In case the setting of `paused` against `rancher-monitoring` is to keep it out of change from any others (e.g., the `fleet-agent`), then here it should be set at very beginning.  then, scale down the `prometheus` or any other deployments  as long as they are downstream obects of `rancher-monitoring`